zerver/lib/onboarding.py

```diff
@@ -220,11 +262,21 @@ def select_welcome_bot_response(human_response_lower: str) -> str:
         )
 
 
+@transaction.atomic
 def send_welcome_bot_response(send_request: SendMessageRequest) -> None:
     """Given the send_request object for a private message from the user
     to welcome-bot, trigger the welcome-bot reply."""
     welcome_bot = get_system_bot(settings.WELCOME_BOT, send_request.message.sender.realm_id)
     human_response_lower = send_request.message.content.lower()
+
+    emoji_reaction = select_emoji_reaction(
+        welcome_bot, human_response_lower, send_request.message.sender
+    )
+    (emoji_code, reaction_type) = emoji_name_to_emoji_code(
+        send_request.message.sender.realm, emoji_reaction
+    )
+    do_add_reaction(welcome_bot, send_request.message, emoji_reaction, emoji_code, reaction_type)
```

Comment:
you're adding a reaction to `send_request.message`, so that's the Message row we should lock. In your current approach you're doing `select_for_update` in the query in `is_first_correct_msg`, which I think isn't correct?

Reply:
Umm. So the only correct approach that comes to mind is that I do something like - 
`message = Message.objects.select_for_update().get(id=send_request.message.id)`
and then replacing the occurrences of `send_request.message` with `message`.

Reply:
Yeah, I think that's reasonable

Reply:
Did it, thanks loads for the review @mateuszmandera!